fearlessTests/Modules/Staking/StakingRewardPayouts/StakingRewardPayoutsTests.swift

```diff
@@ -1,12 +1,170 @@
+            payoutService: payoutServiceThatReturnsError,
+            priceProvider: SingleValueProviderFactoryStub.westendNominatorStub().price,
+            operationManager: OperationManager()
+        )
+        let view = MockStakingRewardPayoutsViewProtocol()
+
+        let viewModelFactory = MockStakingPayoutViewModelFactoryProtocol()
+        let presenter = StakingRewardPayoutsPresenter(
+            chain: .westend,
+            viewModelFactory: viewModelFactory
+        )
+        presenter.interactor = interactor
+        presenter.view = view
+        interactor.presenter = presenter
+
+        // given
+        let viewStateIsLoadingOnPresenterSetup = XCTestExpectation()
+        let viewStateIsNotLoadingWhenPresenterRecievedResult = XCTestExpectation()
+        let viewStateIsErrorWhenPresenterRecievedError = XCTestExpectation()
+
+        stub(view) { stub in
+            when(stub).reload(with: any())
+                .then { viewState in
+                    if case let StakingRewardPayoutsViewState.loading(loading) = viewState, loading {
+                        viewStateIsLoadingOnPresenterSetup.fulfill()
+                    }
+                }
+                .then { viewState in
+                    if case let StakingRewardPayoutsViewState.loading(loading) = viewState, !loading {
+                        viewStateIsNotLoadingWhenPresenterRecievedResult.fulfill()
+                    }
+                }.then { viewState in
+                    if case StakingRewardPayoutsViewState.error(_) = viewState {
+                        viewStateIsErrorWhenPresenterRecievedError.fulfill()
+                    }
+                }
+        }
+
+        // when
+        presenter.setup()
+
+        // then
+        wait(
+            for: [
+                viewStateIsLoadingOnPresenterSetup,
+                viewStateIsNotLoadingWhenPresenterRecievedResult,
+                viewStateIsErrorWhenPresenterRecievedError
+            ],
+            timeout: Constants.defaultExpectationDuration
+        )
+        verify(view, times(3)).reload(with: any())
+        XCTAssert(payoutServiceThatReturnsError.fetchPayoutsCounter == 1)
     }
 
-    override func tearDown() {
-        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    func testShowRewardDetails_whenUserSelectTableRow() {
+        // given
+        let interactor = MockStakingRewardPayoutsInteractorInputProtocol()
+        let wireframe = MockStakingRewardPayoutsWireframeProtocol()
+
+        let viewModelFactory = MockStakingPayoutViewModelFactoryProtocol()
+        let presenter = StakingRewardPayoutsPresenter(
+            chain: .westend,
+            viewModelFactory: viewModelFactory
+        )
+        presenter.wireframe = wireframe
+        presenter.interactor = interactor
+
+        let viewController = StakingRewardPayoutsViewController(presenter: presenter, localizationManager: nil)
```

Comment:
Let's use stub. It might be some issues when we updating view which is not displayed. So for tests we always use view stub